### PR TITLE
Add access list handlers to tshd

### PR DESCRIPTION
**Pull request by gzdunek**
Contributes to https://github.com/gravitational/teleport/issues/32753

This PR adds two new handlers:
* `PromoteAccessRequest`
* `GetSuggestedAccessLists`

Btw, now there is a warning when the app is compiled: 
`[BABEL] Note: The code generator has deoptimised the styling of .../teleport/gen/proto/js/teleport/lib/teleterm/v1/service_pb.js as it exceeds the max of 500KB.`

It looks like it safe to ignore https://github.com/babel/babel/discussions/13676, but we can silence if we want to.

### CODE REVIEW

File: build.assets/genproto.sh [hunk 1]
```diff
@@ -74,7 +74,10 @@ main() {
   [[ $skip_js -eq 0 ]] && echoed buf generate --template=buf-js.gen.yaml \
     --path=proto/prehog/ \
     --path=proto/teleport/lib/teleterm/ \
-    --path=api/proto/teleport/userpreferences/
+    --path=api/proto/teleport/userpreferences/ \
+    --path=api/proto/teleport/accesslist/ \
+    --path=api/proto/teleport/trait/ \
+    --path=api/proto/teleport/header/
```
Comment (gzdunek) [inline build.assets/genproto.sh:80]: These are dependencies of `accesslist`.

File: lib/teleterm/services/accessrequests/accessrequests.go
Comment (gzdunek) [inline lib/teleterm/services/accessrequests/accessrequests.go:1]: I wanted to follow our pattern with adding services, but here the service does almost nothing, so I’m not sure if it is worth keeping.

File: build.assets/genproto.sh (hunk 1 above)
Comment (ravicious) [reply build.assets/genproto.sh:80]: I added #35828 which makes sure that the tests will fail if dependencies are not generated properly.

**Review by ravicious — APPROVED** [review]

File: lib/teleterm/services/accessrequests/accessrequests.go
Comment (ravicious) [reply lib/teleterm/services/accessrequests/accessrequests.go:1]: A remaining issue is that we keep stuffing more and more things into `daemon/daemon.go` as well as expanding our single gRPC service.

I'm still not sure how compelling the reasoning from #14011 is. I keep meaning to talk about this with someone, though I didn't get any response from folks [the last time I asked about this](https://gravitational.slack.com/docs/T040CC4MH/F060VQ9LMFV).

**Review by avatus — APPROVED** [review]

**Pull request merged**